tests/compiletest.rs

```diff
@@ -6,9 +6,4 @@
 fn ui() {
     let check = trybuild::TestCases::new();
     check.compile_fail("tests/ui/*.rs");
-    // Tests that require cargo build instead of cargo check
```

Comment:
These tests were only catching errors in assert_option_safe which now can't fail. The test that was here previously now fails because it doesn't implement OptionTarget instead of the assertion